module/BingoMotion/MotionPerformer.cpp

```diff
@@ -0,0 +1,146 @@
+/**
+ * @file	MotionPerformer.cpp
+ * @brief	動作実行クラス
+ * @author	Hisataka-Hagiyama,uchyam
+ */
+
+#include "MotionPerformer.h"
+
+MotionPerformer::MotionPerformer(LineTracer& _lineTracer)
+  : lineTracer(_lineTracer),
+    inCrossRight(_lineTracer),
+    inCrossLeft(_lineTracer),
+    toCrossMotion(_lineTracer),
+    directionChanger(_lineTracer)
+{
+}
+
+std::vector<MOTION> MotionPerformer::motionLog = {};
+
+void MotionPerformer::runForward()
+{
+  inCrossStraight.runForward();
+  motionLog.push_back(MOTION::RF);
+}
+
+void MotionPerformer::runRight()
+{
+  inCrossRight.runRight();
+  motionLog.push_back(MOTION::RR);
+}
+
+void MotionPerformer::runLeft()
+{
+  inCrossLeft.runLeft();
+  motionLog.push_back(MOTION::RL);
+}
+
+void MotionPerformer::runToCross()
+{
+  toCrossMotion.runToCross();
+  motionLog.push_back(MOTION::RTC);
+}
+
+void MotionPerformer::changeDirection(int angle, bool isClockwise)
+{
+  directionChanger.changeDirection(angle, isClockwise);
+
+  int tmp = std::abs(angle / 45);
+  for(int i = 0; i < tmp; i++) {
+    if(angle > 0) {
+      motionLog.push_back(MOTION::CDC);
+    } else {
+      motionLog.push_back(MOTION::CDRC);
+    }
+  }
+}
+
+void MotionPerformer::setBlockPivotTurn(bool isClockwise)
+{
+  blockPivotTurn.setBlockPivotTurn(isClockwise);
+  if(isClockwise) {
+    motionLog.push_back(MOTION::PSETR);
+  } else {
+    motionLog.push_back(MOTION::PSETL);
+  }
+}
+
+void MotionPerformer::throwBlock(bool isClockwise)
```

Comment:
setBlockPivotTurnと合わせたほうがいいから以下のようにした方がよくない？
だからBlockThrowerクラスの関数名もsetBlockThrowに変更してください
```suggestion
void MotionPerformer::setBlockThrow(bool isClockwise)
```